eqt/ui/UIFormWidget.py

```diff
@@ -226,26 +340,23 @@ def getWidgetState(self, widget, role=None):
         if isinstance(widget, str):
             if role is not None:
                 if role not in ['label', 'field']:
-                    raise ValueError('role must be either "label", "field" or None')
-                name = widget + '_' + role
-            else:
+                    raise ValueError('Role must be either "label", "field" or None')
+                name_role = widget + '_' + role
                 name = widget
+            else:
+                name, role = self._getNameAndRoleFromNameKey(widget)
+                name_role = name + '_' + role
 
             try:
-                widget = self.widgets[name]
+                widget = self.widgets[name_role]
             except KeyError:
-                if role is None:
-                    try:
-                        widget = self.widgets[name + '_field']
-                    except KeyError:
-                        raise KeyError('No widget with name: ' + name + ' or ' + name + '_field')
-                else:
-                    raise KeyError('No widget with name: ' + name)
-
+                        raise KeyError('No widget associated with the dictionary key `'+ name_role)
```

Comment:
I think the indentation is wrong.
```suggestion
                raise KeyError('No widget associated with the dictionary key `'+ name_role)
```

Reply:
Changed.